appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -24,11 +26,16 @@ internal class StateMachine(
 ) {
 
     private val _stateFlow = MutableSharedFlow<State>(1)
-    val stateFlow: SharedFlow<State>
-        get() = _stateFlow
+    val stateFlow: Flow<State>
+        // important that we use .distinctUntilChanged explicitly here, as it is not enabled
+        // by default on SharedFlows (only on StateFlows) - duplicate states can sent
+        // during error transitions, and we don't want those to actually be emitted to listeners
+        get() = _stateFlow.distinctUntilChanged()
```

Comment:
@andretortolano fyi bundled this fix in here as well

it is quite confusing that we have something named "state flow" that is really a SharedFlow 🙃